### DISCUSSION TITLE
Does O3DE support scene dynamic loading function?

### DISCUSSION THREAD

**Discussion by z00249373** [o3de/o3de#4828] [Ideas]
Is there currently a dynamic scene loading function？that is, the player's browsing range is limited, and the naked scene data can not be loaded

If it is not currently supported, is there any plan in the future?

**Comment by forhalle** [comment]
sig/core - You may want to convert this item to a discussion.

**Reply by Infurael** [reply]
It's possible to load any scene data that's stored in a spawnable (typically created from a prefab) through the Spawnable Entities Interface. Note that this is a core system that's currently not exposed to scripting as it's considered work-in-progress. It also only spawns entities so to make a scene work the spawnable will need to have a camera, lights, etc. or those need to be programmatically added.

If you're more interested in loading a level than that can currently be done through a few entry points, but the easiest is probably using `LoadLevel <relative.path.to.level.spawnable>` which takes any spawanble with level information.

Script Canvas has a node to spawn anything from a spawnable, including entire scenes which can also spawn  information.

Work is ongoing to generate spawnables that contain scene information automatically from scene files like fbx. Other work that's ongoing will allow aliases to be setup for entities so it's possible to dynamically have an entity from another spawnable spawn when the original entity is being instanced through the Spawnable Entities Interface.
